RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -0,0 +1,299 @@
+        double replace_inf_value = var_pset.at("replace_inf_value");
+        double lower_bound = var_pset.at("lower_bound");
+        double upper_bound = var_pset.at("upper_bound");
+        double pad = var_pset.contains("pad") ? double(var_pset.at("pad")) : 0;
+        prep_params.var_info_map[var_name] =
+            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, pad);
+      }
+    }
+  } else {
+    // otherwise use the PSet in the python config file
+    const auto &prep_pset = iConfig.getParameterSet("preprocessParams");
+    input_names_ = prep_pset.getParameter<std::vector<std::string>>("input_names");
+    for (const auto &group_name : input_names_) {
+      const auto &group_pset = prep_pset.getParameterSet(group_name);
+      auto &prep_params = prep_info_map_[group_name];
+      prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
+      prep_params.min_length = group_pset.getParameter<unsigned>("var_length");
+      prep_params.max_length = prep_params.min_length;
+      const auto &var_info_pset = group_pset.getParameterSet("var_infos");
+      for (const auto &var_name : prep_params.var_names) {
+        const auto &var_pset = var_info_pset.getParameterSet(var_name);
+        double median = var_pset.getParameter<double>("median");
+        double norm_factor = var_pset.getParameter<double>("norm_factor");
+        double replace_inf_value = var_pset.getParameter<double>("replace_inf_value");
+        double lower_bound = var_pset.getParameter<double>("lower_bound");
+        double upper_bound = var_pset.getParameter<double>("upper_bound");
+        prep_params.var_info_map[var_name] =
+            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, 0);
+      }
+    }
+  }
+  if (debug_) {
+    for (unsigned i = 0; i < input_names_.size(); ++i) {
+      const auto &group_name = input_names_.at(i);
+      if (!input_shapes_.empty()) {
+	std::cout << group_name << "\nshapes: ";
+        for (const auto &x : input_shapes_.at(i)) {
+	  std::cout << x << ", ";
+        }
+      }
+      std::cout << "\nvariables: ";
+      for (const auto &x : prep_info_map_.at(group_name).var_names) {
+	std::cout << x << ", ";
+      }
+      std::cout << "\n";
+    }
+    std::cout << "flav_names: ";
+    for (const auto &flav_name : flav_names_) {
+      std::cout << flav_name << ", ";
+    }
+    std::cout << "\n";
+  }
+
+  // get output names from flav_names
+  for (const auto &flav_name : flav_names_) {
+    produces<JetTagCollection>(flav_name);
+  }
+  //preprocessInfoLoader(&iConfig);
+}
+
+ParticleNetSonicJetTagsProducer::~ParticleNetSonicJetTagsProducer() {}
+
+void ParticleNetSonicJetTagsProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  // pfDeepBoostedJetTags
+  edm::ParameterSetDescription desc;
+  TritonClient::fillPSetDescription(desc);
+  desc.add<unsigned>("batchSize", 1);
```

Comment:
remove this

Reply:
When I remove this line, I get the following error:
----- Begin Fatal Exception 01-Jul-2021 11:30:47 CDT-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=ParticleNetSonicJetTagsProducer label='pfParticleNetSonicJetTags'
Exception Message:
MissingParameter: Parameter 'batchSize' not found.
----- End Fatal Exception -------------------------------------------------

I guess "batchsize" needs to be declared somewhere?  Maybe I'm misinterpreting

Reply:
this is because you hadn't deleted it in the python config yet.

Reply:
(or in the constructor above)